xnat_tools/xnat_utils.py

```diff
@@ -72,16 +72,11 @@ def get_scan_ids(connection, host, session):
     _logger.info("------------------------------------------------")
     _logger.info(f"Get scans.")
     r = get(connection, host + "/data/experiments/%s/scans" % session, params={"format": "json"})
-    scanRequestResultList = r.json()["ResultSet"]["Result"]
+    scanRequestResultList = sorted(r.json()["ResultSet"]["Result"], key=lambda x: int(x['ID']))
     scanIDList = [scan['ID'] for scan in scanRequestResultList]
     seriesDescList = [scan['series_description'] for scan in scanRequestResultList]  # { id: sd for (scan['ID'], scan['series_description']) in scanRequestResultList }
     _logger.debug('Found scans %s.' % ', '.join(scanIDList))
     _logger.debug('Series descriptions %s' % ', '.join(seriesDescList))
-
-    # Fall back on scan type if series description field is empty
-    if set(seriesDescList) == set(['']):
-        seriesDescList = [scan['type'] for scan in scanRequestResultList]
-        _logger.debug('Fell back to scan types %s' % ', '.join(seriesDescList))
```

Comment:
There is no `type` key on the `scan` data. I checked a couple datasets. Am I missing something?

Reply:
I'm not sure ;) - Much on of the API interaction came from this [script](https://bitbucket.org/nrg_customizations/nrg_pipeline_dicomtobids/src/default/scripts/catalog/DicomToBIDS/scripts/dcm2bids_wholeSession.py) which was in the "supported" container for converting to BIDS. 

The docs do seem [docs](https://wiki.xnat.org/display/XAPI/Image+Session+Scans+API) seem to indicate that there is a "type" in the response... but also that's an integer, which doesn't make sense... I'm happy to leave it out.